# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1448660984 18000
#      Fri Nov 27 16:49:44 2015 -0500
# Branch stable
# Node ID 51ab9145f6e677e46e04765ef0f3df894f6277f7
# Parent  e0e55ce86ba0257767821f61625f102465c9e9fb
set all lexical_feedback variables in init function (bug #46522)

* lex.ll (lexical_feedback::init): Set all member variables.

diff --git a/libinterp/parse-tree/lex.ll b/libinterp/parse-tree/lex.ll
--- a/libinterp/parse-tree/lex.ll
+++ b/libinterp/parse-tree/lex.ll
@@ -2017,46 +2017,48 @@ lexical_feedback::reset (void)
   looking_at_matrix_or_assign_lhs = false;
   looking_for_object_index = false;
   looking_at_indirect_ref = false;
   parsing_class_method = false;
   parsing_classdef = false;
   maybe_classdef_get_set_method = false;
   parsing_classdef_get_method = false;
   parsing_classdef_set_method = false;
+  quote_is_transpose = false;
   force_script = false;
   reading_fcn_file = false;
   reading_script_file = false;
   reading_classdef_file = false;
   input_line_number = 1;
   current_input_column = 1;
   bracketflag = 0;
   braceflag = 0;
   looping = 0;
   defining_func = 0;
   looking_at_function_handle = 0;
   block_comment_nesting_level = 0;
+  command_arg_paren_count = 0;
   token_count = 0;
   current_input_line = "";
   comment_text = "";
   help_text = "";
   string_text = "";
   string_line = 0;
   string_column = 0;
   fcn_file_name = "";
   fcn_file_full_name = "";
   looking_at_object_index.clear ();
   looking_at_object_index.push_front (false);
-  command_arg_paren_count = 0;
 
   while (! parsed_function_name.empty ())
     parsed_function_name.pop ();
 
+  pending_local_variables.clear ();
+  symtab_context.clear ();
   nesting_level.reset ();
-
   tokens.clear ();
 }
 
 int
 lexical_feedback::previous_token_value (void) const
 {
   const token *tok = tokens.front ();
   return tok ? tok->token_value () : 0;
